subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/java/tasks/ComparePublications.java

```diff
@@ -2,65 +2,58 @@
 
 import org.gradle.api.logging.Logger;
 import org.gradle.api.logging.Logging;
-import org.gradle.api.publish.maven.tasks.GenerateMavenPom;
 import org.shipkit.gradle.java.ComparePublicationsTask;
-import org.shipkit.internal.comparison.PomComparator;
+import org.shipkit.internal.comparison.ComparePublicationsResultFormatter;
+import org.shipkit.internal.comparison.StringComparator;
 import org.shipkit.internal.comparison.ZipComparator;
 import org.shipkit.internal.comparison.diff.Diff;
+import org.shipkit.internal.gradle.java.ComparePublicationsPlugin;
+import org.shipkit.internal.gradle.util.ZipUtil;
 import org.shipkit.internal.notes.util.IOUtil;
 
 import java.io.File;
-import java.util.ArrayList;
-import java.util.List;
 
 public class ComparePublications {
 
     private final static Logger LOG = Logging.getLogger(ComparePublications.class);
+    public static final String DEPENDENCY_INFO_FILEPATH = "META-INF/" + ComparePublicationsPlugin.DEPENDENCY_INFO_FILENAME;
 
     public void comparePublications(ComparePublicationsTask task) {
-        if (task.getPreviousVersion() == null) {
-            LOG.lifecycle("{} - previousVersion is not set, nothing to compare, skipping", task.getPath());
-            return;
-        }
-        if (!task.getPreviousPom().exists() || !task.getPreviousSourcesJar().exists()) {
+        if (!task.getPreviousSourcesJar().exists()) {
             LOG.lifecycle("{} - previous publications not found, nothing to compare, skipping", task.getPath());
             return;
         }
 
-        GenerateMavenPom pomTask = (GenerateMavenPom) task.getProject().getTasks().getByName(task.getPomTaskName());
-
         //TODO let's add decent validation and descriptive error messages to the user
-        assert pomTask.getDestination().isFile();
         assert task.getSourcesJar().getArchivePath().isFile();
 
-        File currentVersionPomFile = pomTask.getDestination();
         File currentVersionSourcesJarFile = task.getSourcesJar().getArchivePath();
 
-        LOG.lifecycle("{} - about to compare publications, for versions {} and {}",
-                task.getPath(), task.getPreviousVersion(), task.getCurrentVersion());
+        LOG.lifecycle("{} - about to compare publications",
+                task.getPath());
+
+        if (!ZipUtil.fileContainsEntry(task.getPreviousSourcesJar(), DEPENDENCY_INFO_FILEPATH)) {
+            LOG.lifecycle("{} - previous {} file not found, nothing to compare, skipping", task.getPath(), DEPENDENCY_INFO_FILEPATH);
+            return;
```

Comment:
I don't think we should bail out here. What if someone disables dependency-info.json generation? We still want to compare the sources jar. Thoughts?